src/splice.test.ts

```diff
@@ -0,0 +1,29 @@
+import { pipe } from './pipe';
+import { splice } from './splice';
+
+describe('data first', () => {
+  test('splice', () => {
+    expect(splice([1, 2, 3, 4, 5, 6, 7, 8], 2, 3, [])).toEqual([1, 2, 6, 7, 8]);
+    expect(splice([1, 2, 3, 4, 5, 6, 7, 8], 2, 3, [9, 10])).toEqual([
+      1, 2, 9, 10, 6, 7, 8,
+    ]);
+  });
+
+  test('immutability', () => {
+    const data = [1, 2, 3];
+    const result = splice(data, 0, 0, []);
+    expect(result).toEqual(data);
+    assert.notStrictEqual(result, data);
```

Comment:
I think it's easier when all assertions are of the same flavor, we usually use expect so you should use it here too
```suggestion
    expect(result).not.toBe(data);
```

Reply:
I corrected it in this line: https://github.com/remeda/remeda/pull/393/files#diff-0696fb3eb6df2fdeb8e06f4dab9cd586776e32d60d7001dc8eb02cd00ad58061R23